tests/cases/compiler/uncalledFunctionChecksDontWorkWithNegation.ts

```diff
@@ -0,0 +1,16 @@
+// @strictNullChecks: true
+
+declare function isFoo(): boolean;
+declare const isUndefinedFoo: (() => boolean) | undefined;
+
+if (isFoo) {
+    // error
+}
+
+if (!isFoo) {
+    // error
+}
+
+if (!isUndefinedFoo) {
+    // no error
+}
```

Comment:
would also like to see what happens when we issue this error for unawaited promises, like so:
```ts
declare function foo(): Promise<string | undefined>;
const awaitedValue = foo();
if (awaitedValue) { // Emits ts2801
    // ...
}
if (!awaitedValue) { // What happens here?
    // ...
}
```